bpf/lib/nodeport_egress.h

```diff
@@ -306,13 +306,34 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 	void *data, *data_end;
 	struct iphdr *ip4;
 	int l4_off, ret;
+	struct endpoint_info *ep;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 
 	snat_v4_init_tuple(ip4, NAT_DIR_EGRESS, &tuple);
 	l4_off = ETH_HLEN + ipv4_hdrlen(ip4);
 
+	ep = __lookup_ip4_endpoint(ip4->saddr);
+	if (ep && ep->parent_ifindex && ep->parent_ifindex != NATIVE_DEV_IFINDEX) {
```

Comment:
I believe we want to constrain this additional handling to `IS_BPF_HOST`, correct?

Reply:
Ah, yes. This logic shouldn't happen for overlay or wireguard. Had to look it up, I didn't actually know they also called `handle_nat_fwd`